tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/RunTopLevelBalInProjectTestCase.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.packaging;
+
+import org.ballerinalang.test.BaseTest;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.ballerinalang.test.context.Utils;
+import org.ballerinalang.test.utils.PackagingTestUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Testing running a top level bal file inside a project.
+ *
+ * @since 0.982.0
+ */
+public class RunTopLevelBalInProjectTestCase extends BaseTest {
```

Comment:
Whats a top level bal ? I think it can be at any level for which you would normalize the path?

Reply:
I think it is a file in the project root level (parallel to packages).

Reply:
@Kishanthan the issue comes when we are trying to run or build a bal file in the project root level